### PR TITLE
Implement 1 DALARAN card

### PR DESCRIPTION
**Pull request by goltong1**
This revision includes:
- Implement 1 DALARAN card
    - 'Spellbook Binder' (DAL_089)
- Implement method for self condition
    - `HasSpellPower()`: SelfCondition wrapper for checking the player has spell power

### CODE REVIEW

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Sources/Rosetta/CardSets/DalaranCardsGen.cpp  1
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4972346&bid=16424368)

**Comment by utilForever** [comment]
Sorry. I'll review it tomorrow. :)

File: Documents/CardList.md [hunk 1]
```diff
@@ -480,14 +480,14 @@ DALARAN | DAL_350 | Crystal Power |
...
-DALARAN | DAL_355 | Lifeweaver |  
+DALARAN | DAL_355 | Lifeweaver | 
 DALARAN | DAL_357 | Lucentbark |  
 DALARAN | DAL_366 | Unidentified Contract |  
 DALARAN | DAL_371 | Marked Shot |  
 DALARAN | DAL_372 | Arcane Fletcher |  
 DALARAN | DAL_373 | Rapid Fire |  
 DALARAN | DAL_376 | Oblivitron |  
-DALARAN | DAL_377 | Nine Lives |  
+DALARAN | DAL_377 | Nine Lives |
```
Comment (utilForever) [inline Documents/CardList.md]: 불필요한 공백 없애주세요.

File: Documents/CardList.md [hunk 2]
```diff
@@ -480,14 +480,14 @@ DALARAN | DAL_350 | Crystal Power |
 DALARAN | DAL_351 | Blessing of the Ancients | O
 DALARAN | DAL_352 | Crystalsong Portal |  
 DALARAN | DAL_354 | Acornbearer |  
-DALARAN | DAL_355 | Lifeweaver |  
+DALARAN | DAL_355 | Lifeweaver |
```
Comment (utilForever) [inline Documents/CardList.md]: Lifeweaver 카드 구현되어 있으니 O로 표시해주세요.

File: Sources/Rosetta/Conditions/SelfCondition.cpp [hunk 3]
```diff
@@ -197,6 +197,16 @@ SelfCondition SelfCondition::HasReborn()
...
 }
 
+SelfCondition SelfCondition::HasSpellPower()
+{
+    return SelfCondition([=](Playable* playable) -> bool {
+        if (playable->player->currentSpellPower > 0)
+        {
+            return true;
+        }
+        return false;
```
Comment (utilForever) [inline Sources/Rosetta/Conditions/SelfCondition.cpp]: `return playable->player->currentSpellPower > 0`으로 바꿀 수 있겠네요.

File: Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp [hunk 4]
```diff
@@ -471,6 +471,48 @@ TEST(NeutralDalaranTest, DAL_085_DalaranCrusader)
...
+    opPlayer->SetTotalMana(10);
+    opPlayer->SetUsedMana(0);
+
+    const auto card1 =
+        Generic::DrawCard(curPlayer, Cards::FindCardByName("Kobold Geomancer"));
+    const auto card2 =
+        Generic::DrawCard(curPlayer, Cards::FindCardByName("Spellbook Binder"));
+    game.Process(curPlayer, PlayCardTask::Minion(card1, 1));
+    game.Process(curPlayer, PlayCardTask::Minion(card2, 2));
+    EXPECT_EQ(curPlayer->currentSpellPower,1);
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp]: ,와 1 사이에 빈칸 넣어주세요.

File: Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp [hunk 5]
```diff
@@ -471,6 +471,48 @@ TEST(NeutralDalaranTest, DAL_085_DalaranCrusader)
...
+    EXPECT_EQ(curPlayer->GetHandZone()->GetCount(), 5);
+}
```
Comment (utilForever) [inline Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp]: 주문 공격력이 없을 때 카드를 안뽑는지 확인할 수 있는 테스트 코드도 추가되면 좋겠네요.

Comment (goltong1) [reply Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp]: 리뷰 사항 전부 수정했습니다. 확인바랍니다.

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Sources/Rosetta/CardSets/DalaranCardsGen.cpp  1
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4972346&bid=16424368)

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Sources/Rosetta/CardSets/DalaranCardsGen.cpp  3
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4972346&bid=16424368)

**Comment by utilForever** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- Sources/Rosetta/CardSets/DalaranCardsGen.cpp  3
- Tests/UnitTests/CardSets/DalaranCardsGenTests.cpp  22
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4972346&bid=16424368)

**Pull request merged**